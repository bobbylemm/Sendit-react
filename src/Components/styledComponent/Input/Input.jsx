import styled from 'styled-components';

const Input = styled.input`
  width: 100%;
  line-height: 1.3;
  padding: 1rem;
  margin-bottom: 1.9rem;
  height: 3.5rem;
  border: 0.1rem solid #b3aaaa;
  border-radius: 0.5rem;
  cursor: auto;
  display: inline-block;
  outline: none;
}
::placeholder {
  opacity: 0.5;
}
`;

export default Input;
