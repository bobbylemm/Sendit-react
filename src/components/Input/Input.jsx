import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Input from '../styledComponent/Input/Input';

const InputField = ({
  input,
  placeholder,
  type,
  name,
  id,
  meta: { error, touched }
}) => (
  <Fragment>
    <Input
      {...input}
      type={type}
      placeholder={placeholder}
      name={name}
      id={id}
    />
    {touched && error && <span className="error-message">{error}</span>}
  </Fragment>
);
InputField.propTypes = {
  input: PropTypes.object,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  meta: PropTypes.object
};

export default InputField;
