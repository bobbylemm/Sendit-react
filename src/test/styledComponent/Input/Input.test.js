import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from '../../../components/styledComponent/Input/Input';

describe('testing the button element', () => {
  it('should mount button', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('border', '0.1rem solid #b3aaaa');
  });
});
