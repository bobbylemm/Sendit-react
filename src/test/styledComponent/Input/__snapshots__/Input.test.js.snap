// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the button element should mount button 1`] = `
.c0 {
  width: 100%;
  line-height: 1.3;
  padding: 1rem;
  height: 3.5rem;
  border: 0.1rem solid #b3aaaa;
  border-radius: 0.5rem;
  cursor: auto;
  display: inline-block;
  outline: none;
}

.c0::-webkit-input-placeholder {
  opacity: 0.5;
}

.c0::-moz-placeholder {
  opacity: 0.5;
}

.c0:-ms-input-placeholder {
  opacity: 0.5;
}

.c0::placeholder {
  opacity: 0.5;
}

<input
  className="c0"
/>
`;
